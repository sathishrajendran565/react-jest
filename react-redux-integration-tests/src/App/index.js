import { ConnectedRouter } from 'connected-react-router';
import React from 'react';
import { Provider } from 'react-redux';
import { Link } from 'react-router-dom';
import history from '../history';
import Routes from '../Routes';
import store from '../store';
import './index.css';

export default function App() {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <div className="app">
          <h1>BEST APP EVER</h1>
          <Link to="/choose-your-pokemon">Choose Your Pokemon</Link>
          <Routes />
        </div>
      </ConnectedRouter>
    </Provider>
  );
}
