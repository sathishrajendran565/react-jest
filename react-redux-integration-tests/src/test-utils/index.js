import { ConnectedRouter } from 'connected-react-router';
import { mount } from 'enzyme';
import { mergeDeepRight } from 'ramda';
import React from 'react';
import { Provider } from 'react-redux';
import history from '../history';
import rootReducer from '../reducer';
import { createStoreWithMiddleWare } from '../store';

export const mockData = {
  pokemon: {
    id: 26,
    species: { name: 'raichu' },
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/26.png'
    }
  }
};

export const makeMountRender = (Component, defaultProps = {}) => {
  return (customProps = {}) => {
    const props = {
      ...defaultProps,
      ...customProps
    };
    return mount(<Component {...props} />);
  };
};

export const makeStore = (customState = {}) => {
  const root = rootReducer({}, { type: '@@INIT' });
  const state = mergeDeepRight(root, customState);

  return createStoreWithMiddleWare(rootReducer, state);
};

export const reduxify = (Component, props = {}, state = {}) => {
  return function reduxWrap() {
    return (
      <Provider store={makeStore(state)}>
        <ConnectedRouter history={history}>
          <Component {...props} />
        </ConnectedRouter>
      </Provider>
    );
  };
};

export const snapshotify = reactWrapper => {
  return reactWrapper.html();
};

export const mocker = apiMock => ({
  fetchRandomPokemon() {
    apiMock
      .onGet(/https:\/\/pokeapi.co\/api\/v2\/pokemon\/\d+/)
      .reply(config => {
        const numberCheck = RegExp(/pokemon\/(\d+)/);
        const pokemonInt = Number(numberCheck.exec(config.url)[1]);

        if (pokemonInt > 151) {
          return [500];
        }

        return [200, mockData.pokemon];
      });

    return this;
  }
});

export const ticks = (callbacks = []) => {
  if (callbacks.length < 1) {
    return null;
  }

  setTimeout(() => {
    const cb = callbacks.shift();

    if (cb) {
      cb();
    }

    ticks(callbacks);
  });
};
