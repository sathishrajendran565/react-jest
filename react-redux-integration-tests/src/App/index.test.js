import { makeMountRender, reduxify, snapshotify } from '../test-utils';
import App from './';

describe('<App />', function() {
  it('matches snapshot', function() {
    const wrapper = makeMountRender(reduxify(App))();
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });
});
