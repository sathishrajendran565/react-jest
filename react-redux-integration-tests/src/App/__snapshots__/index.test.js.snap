// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div class="app">
  <h1>
    BEST APP EVER
  </h1>
  <a href="/choose-your-pokemon">
    Choose Your Pokemon
  </a>
</div>
`;
